### PR TITLE
[clang] Extend clang's <limits.h> to define *LONG_LONG*_ macros for bionic

### PR DESCRIPTION
**Pull request by ZijunZhaoCCK**
*LONG_LONG*_ macros are not GNU-only extensions any more. Bionic also defines them.

### CODE REVIEW

File: clang/lib/Headers/limits.h [hunk 1]
```diff
@@ -111,11 +111,12 @@
 #define ULLONG_MAX (__LONG_LONG_MAX__*2ULL+1ULL)
 #endif
 
-/* LONG_LONG_MIN/LONG_LONG_MAX/ULONG_LONG_MAX are a GNU extension.  It's too bad
-   that we don't have something like #pragma poison that could be used to
-   deprecate a macro - the code should just use LLONG_MAX and friends.
+/* LONG_LONG_MIN/LONG_LONG_MAX/ULONG_LONG_MAX are a GNU extension. Bionic also
```
Comment (pirama-arumuga-nainar) [inline clang/lib/Headers/limits.h:114]: Maybe "Android's bionic also ..."

**Review by pirama-arumuga-nainar — APPROVED** [review]
LGTM for Android; tagging @AaronBallman as well.

**Review by AaronBallman — COMMENTED** [review]
Release note and test coverage?

**Comment by ZijunZhaoCCK** [comment]
> Release note and test coverage?

I tested it locally. I copied code to `prebuilt/.../limits.h` in aosp and removed bionic's copy and rebuilt to see whether it works. And it worked well. Any extra upstream test coverage needed?

**Comment by AaronBallman** [comment]
> > Release note and test coverage?
> 
> I tested it locally. I copied code to `prebuilt/.../limits.h` in aosp and removed bionic's copy and rebuilt to see whether it works. And it worked well. Any extra upstream test coverage needed?

Yes, we've got tests for Clang-supplied headers in `clang/test/Headers`; you should pass `-ffreestanding` on the `RUN` line and that should find Clang's headers rather than the system headers.

File: clang/docs/ReleaseNotes.rst [hunk 2]
```diff
@@ -319,6 +319,7 @@ C23 Feature Support
 
 - Clang now supports `N3029 <https://www.open-std.org/jtc1/sc22/wg14/www/docs/n3029.htm>`_ Improved Normal Enumerations.
 - Clang now officially supports `N3030 <https://www.open-std.org/jtc1/sc22/wg14/www/docs/n3030.htm>`_ Enhancements to Enumerations. Clang already supported it as an extension, so there were no changes to compiler behavior.
+- Extend clang's <limits.h> to define *LONG_LONG*_ macros for Android's bionic.
```
Comment (AaronBallman) [inline clang/docs/ReleaseNotes.rst:322]:
```suggestion
- Extend Clang's ``<limits.h>`` to define ``LONG_LONG_*`` macros for Android's bionic.
```
Also, this should be moved up under the `C Language Changes` heading (it's currently listed under `C23 Feature Support`).

**Review by AaronBallman — APPROVED** [review]
LGTM aside from some minor changes to the release note.

**Review by enh-google — APPROVED** [review]

**Pull request merged**